packages/components/src/compare-filter/index.js

```diff
@@ -27,10 +27,15 @@ export class CompareFilter extends Component {
 		this.state = {
 			selected: [],
 		};
-
 		this.clearQuery = this.clearQuery.bind( this );
 		this.updateQuery = this.updateQuery.bind( this );
 		this.updateLabels = this.updateLabels.bind( this );
+		this.onClickHandlers = [ this.updateQuery ];
```

Comment:
Rather than using an array here it would be simpler and more understandable to just call `this.updateQuery` and `this.props.onClick` directly in `onButtonClicked()`

Reply:
Thank you for the suggestion. Updated with commit https://github.com/woocommerce/woocommerce-admin/pull/5649/commits/987f700984ad1bb3406255b395558057b9319605